packages/core/src/editor/BlockNoteExtensions.ts

```diff
@@ -265,14 +270,72 @@ const getTipTapExtensions = <
 
         const nonbreakingSpace1 = document.createTextNode("\u2060");
         const nonbreakingSpace2 = document.createTextNode("\u2060");
-        cursor.insertBefore(nonbreakingSpace1, null);
-        cursor.insertBefore(label, null);
-        cursor.insertBefore(nonbreakingSpace2, null);
+
+        let hideTimeout: NodeJS.Timeout | undefined = undefined;
+        let oldDoc = opts.editor.document;
+        const awareness = opts.collaboration!.provider.awareness as Awareness;
+
+        awareness.on(
+          "change",
+          (a: {
+            added: Array<number>;
+            updated: Array<number>;
+            removed: Array<number>;
+          }) => {
+            if (!a.updated.includes(user.clientID)) {
+              return;
+            }
+
+            if (hideTimeout) {
+              clearTimeout(hideTimeout);
+            }
+
+            if (
+              JSON.stringify(opts.editor.document) !== JSON.stringify(oldDoc)
+            ) {
```

Comment:
This can be expensive with large documents. Maybe a deep equal algo, Or if these are prosemirror nodes their .eq method

Reply:
We ended up showing the labels on selection changes as well as content changes, so removed this altogether